games/main.tex

```diff
@@ -0,0 +1,1123 @@
+as arguments the provider address, the plaintiff address, and the note id (the hash of the signed note)
+representing the service request.
+The \gloss{swindle} contract orchestrates the transition through the stages calling the witness contracts,
+handles grace periods to control the deadline for submitting evidence (challenges or their refutations) to
+eventually reach a guilty/non-guilty verdict on the grounds of witness testimonies.
+In case of a guilty verdict, the deposit sworn on is forfeited and the peer's registration is
+terminated.
+
+\begin{center}
+\begin{figure}
+\begin{center}
+\begin{tikzpicture}[->,>=stealth',shorten >=1pt,auto,node distance=2c m,
+                    semithick]
+  \tikzstyle{witness}=[fill=orange!50,draw]
+  % \tikzstyle{every state}=[fill=gray!10,draw]
+
+  \node[initial,state] (S)                    {$S$};
+  \node[state,witness]         (W0_1) [above right of=S] {$w_0$};
+  \node[state,witness]         (W0_2) [below right of=S] {$w_0$};
+  \node[state,fill=green!60]         (N/G) [below right of=W0_1] {$N/G$};
+  \node[state,witness]         (W1) [above right of=N/G] {$w_1$};
+  \node[state,witness]         (W2) [below right of=N/G] {$w_2$};
+  \node[state,witness]         (W3) [below right of=W1] {$w_3$};
+  \node[state,fill=red!60]         (G) [right of=W3]       {$G$};
+
+  \path (S) edge   node {$A_0$} (W0_1)
+            edge   node {$A_1$} (W0_2)
+        (W0_1) edge [loop above] node {n/a} (W0_1)
+            edge [red,thick]             node {T} (W1)
+            edge [green,thick]             node {F} (N/G)
+        (W0_2) edge [loop below] node {n/a} (W0_2)
+            edge [red,thick]             node {T} (W2)
+            edge [green,thick]             node {F} (N/G)
+        (W1) edge [loop above] node {n/a} (W1)
+            edge [red,thick]             node {T} (W3)
+            edge [green,thick]             node {F} (N/G)
+        (W2) edge [loop below] node {n/a} (W2)
+            edge  [red,thick]            node {T} (W3)
+            edge  [green,thick]            node {F} (N/G)
+        (N/G) edge              node {*} (S)
+        (W3) edge [loop below] node {n/a} (W3)
+            edge  [green,thick]            node {F} (N/G)
+            edge [red,thick]  node {T} (G);
+\end{tikzpicture}
+\end{center}
+\caption{Trial stages represented as a finite state automaton describing a service contract.
+Upon receiving an accusation ($A_0$ or $A_1$) in a transaction, the case contract moves to the first litigation state.
+The orange circles represent states where a witness contract is called, the following state depends on
+the outcome of the witness testimony (the return value of the $\mathit{testimonyFor}$ function call).
+The other nodes call the internal $\mathit{process}$ function: to pay compensation if the challenge is refuted (N/G) or enforce
+the sentence in case of a guilty verdict (G).}
+\label{fig:servicecontract}
+\end{figure}
+\end{center}
+
```

Comment:
When producing the pdf, the figure can be shown on the next page, after the next chapter already started on the previous page. It would be good to reference the Figure in the text itself, e.g. "Refer to figure 7 for an illustration of the Trial process"